src/org/exist/collections/Collection.java

```diff
@@ -2305,4 +2303,32 @@ public QNameRangeIndexSpec getIndexByQNameConfiguration(final DBBroker broker, f
         final IndexSpec idxSpec = getIndexConfiguration(broker);
         return (idxSpec == null) ? null : idxSpec.getIndexByQName(qname);
     }
+
+    @Override
+    public ResourceMetadata getMetadata() {
+        return new CollectionMetadata();
+    }
+
+    class CollectionMetadata implements ResourceMetadata {
+
+        @Override
```

Comment:
Well I feel that a MimeType (or rather Internet Media Type) is something that you assign as an indicator to the representation of a transferable/serialized Resource, however a Collection is not something that you can transfer or serialize. Perhaps better to have `public Optional<String> getMimeType()` then?